### ISSUE TITLE
terraform aws waf should support IPSetForwardedIPConfig attribute of IPSetReferenceStatement

### ISSUE THREAD

**Issue by munishsbs** [terraform-providers/terraform-provider-aws#14465]
Currently `IPSetReferenceStatement` is supported via terraform tag `ip_set_reference_statement` and has only one attribute support which is `arn`. However, it should also have support for non optional tag for `IPSetForwardedIPConfig`. It could be named something like `ip_set_forwarded_ip_config`

**Comment by anGie44** [comment]
Hi @munishsbs, thank you for submitting this issue! We'll first need to update our aws-sdk-go dependency to atleast `v1.33.5` (https://github.com/aws/aws-sdk-go/pull/3416) to support this param w/in the `ip_set_reference statement` as we're currently on `v1.32.12`. Luckily #14000 will include these API changes and will unblock development efforts to address this update 😃

**Comment by adamhathcock** [comment]
Since #14000 is merged, really need this for using WAFv2 with CloudFront as the forwarded IP is needed for IP whitelisting for some reason.

**Comment by adamhathcock** [comment]
Supported on https://github.com/terraform-providers/terraform-provider-aws/issues/14480 which might be a workaround

**Comment by eboulfakhr** [comment]
Hello!

> Supported on #14480 which might be a workaround  

The configuration block for rate-limiting is supported in #14480 but not for IP filtering which is a different statement:
````
statement {
        ip_set_reference_statement {
          arn = aws_wafv2_ip_set.ipset_allow.arn
          ip_set_forwarded_ip_config {
            header_name = "X-Forwarded-For"
            fallback_behavior = "NO_MATCH"
            position = "FIRST"
          }
        }
      }
````
should generate:
````
      "Statement": {
        "IPSetReferenceStatement": {
          "ARN": "arn:aws:wafv2:...",
          "IPSetForwardedIPConfig": {
            "HeaderName": "X-Forwarded-For",
            "FallbackBehavior": "NO_MATCH",
            "Position": "FIRST"
          }
        }
      }
````
while it provokes `Blocks of type "ip_set_forwarded_ip_config" are not expected here.`error (aws provider v3.3.0)

**Comment by anGie44** [comment]
This feature has been merged and will release with  `v3.5.0` of the Terraform AWS Provider, expected out later this week.
